### PR TITLE
Remove .pkl

### LINKED ISSUE #754: Remove itr_0.pkl from root

### PR DESCRIPTION
**Pull request by ahtsan**
Removed the line which creates `itr_0.pkl` and created a pickle test for `garage.tf.optimizers.FiniteDifferenceHvp`. Also added `.pkl` in `.gitignore`.

This will close #754.

### CODE REVIEW

**Comment by krzentner** [comment]
Wait, does this change actually delete the file? I don't see it in the Files Changed list.

File: .gitignore [hunk 1]
```diff
@@ -145,3 +145,6 @@ data/
 garage/config_personal.py
 sandbox/
 MUJOCO_LOG.TXT
+
+# .pkl
```
Comment (ryanjulian) [inline .gitignore]: this doesn't need its own section IMO

**Comment by krzentner** [comment]
> 
> 
> Please delete the existing `iter_0.pkl` file.

Nevermind, I did this by rewriting history.

**Comment by krzentner** [comment]
This PR is now trying to add `iter_0.pkl` back, so please remove that.

**Pull request closed without merging**